src/Http/ApiResponseDataFactory.php

```diff
@@ -18,9 +19,17 @@ public function createFromResponse(DataResponse $response): ApiResponseData
                 ->setErrorMessage($this->getErrorMessage($response));
         }
 
+        $data = $response->getData();
+
+        if (empty($data)) {
```

Comment:
Empty array and null are different response. Just throw exception if data not array and not null, not modify data.